src/components/Timer.vue

```diff
@@ -0,0 +1,37 @@
+<template>
+    <div class="timer">
+        <div class="timer-countdown">{{countdown}}</div>
+        <div class="timer-progress" :style="{width: `${progress}%`}"></div>
+    </div>
+</template>
+
+<script lang="ts">
+    import {defineComponent, ref} from "vue";
+    
+    export default defineComponent({
+        props: {
+            time: {type: Object, required: true}
+        },
+        setup(props) {
+            const countdown = ref("60.00");
+            const progress = ref(100);
```

Comment:
Any reason do you do this with refs rather than simply adding this as data property?

Reply:
ref returns an observable object and is used for two-way data binding to update the view.

Reply:
Well... but you are not doing two-way data binding, right?

Reply:
We do, it's rendered in the template above and won't update without it.

Reply:
I meant it like this: https://jsfiddle.net/api/post/library/pure/#&togetherjs=c6tacD1jpt. The timer will be likely updated from the GameState and not from the UI, right?

Reply:
404... we can calculate the time the timer should end instead of updating the timer through properties all the time.

Reply:
hm... does this work: https://jsfiddle.net/ofcb9spq/? Anyway, https://paste.centos.org/view/210c0516 is probably better explaining it. That way you only have a single value you need to change while the others update declaratively.